readalongs/text/make_jsgf.py

```diff
@@ -40,14 +40,14 @@
 """
 
 
-def make_jsgf(xml, filename, unit="m"):
+def make_jsgf(word_elements, filename="'in-memory'"):
```

Comment:
Any objection to `make_jfsg()` being moved to `make_fsg.py` now that each function doesn't need its own `go()` and `__main__` block? And how is a JSGF different from an FSG? And why JSGF and not JFSG? Oh, wait, Google tells me JSGF stands for Java Speech Grammar Format... We better fix the docstring for this thing, too!

Reply:
nope, go for it!

Reply:
Seems like a good idea. JSGF is used here because the FSG format is an ad-hoc Sphinx thing, and SoundSwallower purposefully doesn't support parsing it from its API, but in actual fact we ought to just create an FSG directly if we don't need to save the temporary file for any reason, as there's much less overhead.

Reply:
I'm guessing right now we're generating the JSGF format in memory and passing its contents as a text variable that SS.js then parses. It doesn't go to disk, but you're right, it's pointless generation and parsing that could be optimized away if the data structure could be passed directly...
Although no, I'm half wrong: the text format is generated by the python web API and received by the Studio web app via a REST call, so encoding for transfer is required no matter what. If we had a closer structure, we'd still need to JSONify it to return it over HTTPS, and I'm not convinced that would be any more efficient...